docs/architecture/message-protocol.md

```diff
@@ -1,65 +1,67 @@
-# Message protocol
+# Message Protocol
 
 ## Why it exists
-Usually communication between server and worker uses events and commands: events go from server to worker,
-worker process them and generates commands that go back to server. Events are attached to Workflow Task, which
-worker gets as response to `PollWorkflowTask` API call, and worker sends commands back
-when it completes Workflow Task with `RespondWorkflowTaskCompleted` API. Workflow Task works as transport on RPC level.
+Usually, communication between the server and the worker uses events and commands: events go from
+the server to the worker, the worker processes them and generates commands that go back to server.
+The events are attached to the Workflow Task, which the worker receives from the `PollWorkflowTask`
+API call, and the worker sends commands back when it completes the Workflow Task with the 
+`RespondWorkflowTaskCompleted` API.
 
-Unfortunately, this way or communication didn't work for Workflow Update. Server can't use events
-to ship Update request to the worker, because worker might reject Update, and it must completely disappear.
-Because history is immutable, server can't delete events from it. Initial implementation
-was using transient event with Update request, which wasn't written to history. This implementation
-was proven to be error-prone and hard to handle on the SDK side. Commands that go back from worker to server
-also can't be used for Update because some SDKs assume that every command will produce exactly one event, 
-which is not true for Update rejections that don't produce any events.
+Unfortunately, this protocol didn't work for Workflow Update. The server cannot use events to ship
+Update request to the worker because in case the Update is rejected, it must completely disappear.
+But because the history is immutable, the server cannot delete any events from it. The initial
+implementation was using a transient (not written to history) event instead, but that implementation
+proved to be error-prone and hard to handle on the SDK side. Similarly, commands can't be used
+for Update either because some SDKs assume that every command will produce *exactly* one event, 
+which is not true for Update rejections as they don't produce an event.
 
-Another protocol was required to implement Workflow Update. Messages are attached to Workflow Task and go in
-both directions, similar to events and commands but don't have limitations listed above.
+Another protocol was required to implement Workflow Update: Messages are attached to Workflow Task
```

Comment:
```suggestion
Another protocol was required to implement Workflow Update: messages are attached to Workflow Task
```
it seems like we don't capitalized "message" as well as "event" although public docs do.

Reply:
I thought you capitalize after a colon, but it doesn't seem to be necessarily true.